deploy/examples/create-external-cluster-resources.py

```diff
@@ -692,8 +698,22 @@ def get_ceph_external_mon_data(self):
             raise ExecutionFailureException(
                 "Only 'v2' address type is enabled, user should also enable 'v1' type as well"
             )
-        ip_port = str(q_leader_details["public_addr"].split("/")[0])
-        return f"{str(q_leader_name)}={ip_port}"
+        ip_addr = str(q_leader_details["public_addr"].split("/")[0])
+
+        if self._arg_parser.v2_port_enable:
+            if len(q_leader_addrvec) > 1:
+                if q_leader_addrvec[0]["type"] == "v2":
+                    ip_addr = q_leader_addrvec[0]["addr"]
+                    return f"{str(q_leader_name)}={ip_addr}"
+                elif q_leader_addrvec[1]["type"] == "v2":
+                    ip_addr = q_leader_addrvec[1]["addr"]
+                    return f"{str(q_leader_name)}={ip_addr}"
+
+            sys.stderr.write(
```

Comment:
this can be third else

Reply:
The thing is, we would have to duplicate this statement if you see as we should shwoing this error if the `len < 1` and if the `type == v2` is not present in both 0th,1th position (though is that situation ever possible?). That's the reason I updated the code a bit to handle both the situations.

Reply:
Make sense 

So that can be a else for `            if len(q_leader_addrvec) > 1: `

            if len(q_leader_addrvec) > 1:
                code 
             else 
                 error

Reply:
Ok, so that means we would never get into a situation where the `v2` type is not present in both 0,1th position, is it?

Reply:
correct

Reply:
Ok, then I will update the code

Reply:
Done